ood-portal-generator/spec/fixtures/ood-portal.conf.dex-full

```diff
@@ -79,7 +79,7 @@
 
   # OIDC configuration
   #
-  OIDCProviderMetadataURL https://example.com:5554/.well-known/openid-configuration
+  OIDCProviderMetadataURL https://example-proxy.com:5554/.well-known/openid-configuration
```

Comment:
I believe this is the right behaviour that we want, @treydock thoughts?  If they have things set behind a proxy they'd need to open this port anyhow, then the well known configuration url is the proxy.

I guess it's a question of network connectivity. Do you think it's ever the case that the server behind the proxy can't reach it (maybe even on this strange port)?

Reply:
The URL you get for `MetadataURL` will depend on the value for `issuer` so it looks like if `proxy_server` is set that modifies `servername` which is used to build `issuer`.  What will happen here is OnDemand mod_auth_openidc will end up having to go through the proxy, which might not be great behavior but should still work. 

Unless I'm missing something, normally things behind a proxy don't talk to the proxy. The proxy, like HAProxy, will communicate downstream to OnDemand and Dex but never the other direction, normally things downstream of a proxy don't necessarily know they are behind a proxy. The Proxy will usually check the downstream services are healthy and what not and route traffic accordingly.  This is a weird situation where mod_auth_openidc will get routed through the proxy back to Dex behind the proxy to download the OpenID configuration.

Reply:
Sorry, so is it OK to have `OIDCProviderMetadataURL` be the proxy? or should it actually be the server?

I'm guessing both may work (unless `mod_auth_openidc` does some check on issuer and metadata hostnames all being the same).

I guess I'm just wondering what the right thing is because we don't have any way to test this out ourselves do we?

Reply:
I think the metadata URL being the proxy should be fine. There might be more involved but I believe that URL is just used to pull down the OIDC data by mod_auth_openidc and probably better to have it match the issuer than not match as I am not sure if there are issues with a mismatch.

Correct, we have no way to test this.